dotCMS/src/main/java/com/dotcms/rest/api/v1/container/ContainerResource.java

```diff
@@ -92,24 +120,318 @@ public ContainerResource(final WebResource webResource,
-            return this.paginationUtil.getPage(request, user, filter, page, perPage, orderBy,
-                    OrderDirection.valueOf(direction), extraParams);
+            final Map<String, Object> extraParams = Maps.newHashMap();
+            extraParams.put(ContainerPaginator.HOST_PARAMETER_ID, (Object) hostId);
+            return this.paginationUtil.getPage(request, user, filter, page, perPage, orderBy, OrderDirection.valueOf(direction),
+                    extraParams);
         } catch (Exception e) {
             Logger.error(this, e.getMessage(), e);
             return ExceptionMapperUtil.createResponse(e, Response.Status.INTERNAL_SERVER_ERROR);
 
         }
     }
-}
\ No newline at end of file
+
+    @GET
+    @JSONP
+    @NoCache
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces({MediaType.APPLICATION_JSON, "application/javascript"})
+    @Path("/{containerId}/content/{contentletId}")
+    public final Response containerContent(@Context final HttpServletRequest req, @Context final HttpServletResponse res,
+            @PathParam("containerId") final String containerId, @PathParam("contentletId") final String contentletId)
+            throws DotDataException, DotSecurityException, ParseErrorException, MethodInvocationException,
+            ResourceNotFoundException, IOException, IllegalAccessException, InstantiationException, InvocationTargetException,
+            NoSuchMethodException {
+
+        final InitDataObject initData = webResource.init(true, req, true);
+        final User user = initData.getUser();
+        PageMode mode = PageMode.get(req);
+        Language id = WebAPILocator.getLanguageWebAPI()
+            .getLanguage(req);
+
+
+        ShortyId contentShorty = APILocator.getShortyAPI()
+            .getShorty(contentletId)
+            .orElseGet(() -> {
+                throw new ResourceNotFoundException("Can't find contentlet:" + contentletId);
+            });
+
+
+
+        final Contentlet contentlet = (contentShorty.subType == ShortType.CONTENTLET) ? APILocator.getContentletAPI()
+            .find(contentShorty.longId, user, mode.showLive)
+                : APILocator.getContentletAPI()
+                    .findContentletByIdentifier(contentShorty.longId, mode == PageMode.ANON, id.getId(), user,
+                            mode == PageMode.ANON);
+
+        if (contentlet.getContentType()
+            .baseType() == BaseContentType.WIDGET) {
+            Map<String, String> response = new HashMap<>();
+            response.put("render", WidgetResource.parseWidget(req, res, contentlet));
+
+            return Response.ok(response)
+                .build();
+
+
+        }
+
+        ShortyId containerShorty = APILocator.getShortyAPI()
+            .getShorty(containerId)
+            .orElseGet(() -> {
+                throw new ResourceNotFoundException("Can't find Container:" + containerId);
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "Can't find Container:" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)